src/cli.rs

```diff
@@ -523,6 +523,15 @@ pub struct Opts {
     )]
     pub pattern: String,
 
+    /// Additional search patterns to match a'la AND (a regular expression, unless '--glob' is used; optional)
+    #[arg(
+        long = "expr",
+        value_name = "pattern",
+        long_help = "(Additionally) filter search results by all patterns via AND. Multiple \
+                     additionalpatterns can be specified."
+    )]
```

Comment:
I would appreciate if we keep this out of the short help text for now (`hide_short_help = true`).